WooCommerce/Classes/POS/ViewModels/ItemListViewModel.swift

```diff
@@ -42,25 +27,135 @@ final class ItemListViewModel: ObservableObject {
         do {
             state = .loading
             items = try await itemProvider.providePointOfSaleItems()
-            state = .loaded
+            if items.count == 0 {
+                let itemListEmpty = EmptyModel(title: Constants.emptyProductsTitle,
+                                                  subtitle: Constants.emptyProductsSubtitle,
+                                                  hint: Constants.emptyProductsHint,
+                                                  buttonText: Constants.emptyProductsButtonTitle)
+                state = .empty(itemListEmpty)
+            } else {
+                state = .loaded(items)
+            }
         } catch {
             DDLogError("Error on load while fetching product data: \(error)")
-            state = .error
+            let itemListError = ErrorModel(title: Constants.failedToLoadTitle,
+                                      subtitle: Constants.failedToLoadSubtitle,
+                                      buttonText: Constants.failedToLoadButtonTitle)
+            state = .error(itemListError)
         }
     }
 
     @MainActor
     func reload() async {
         do {
+            // TODO:
+            // Resolve duplication with populatePointOfSaleItems()
             state = .loading
             let newItems = try await itemProvider.providePointOfSaleItems()
-            // Only clears in-memory items if the `do` block continues, otherwise we keep them in memory.
-            items.removeAll()
-            items = newItems
-            state = .loaded
+            if newItems.count == 0 {
+                let itemListEmpty = EmptyModel(title: Constants.emptyProductsTitle,
```

Comment:
Should we create these directly inside of the struct? Sort of
```
func copy() -> EmptyModel { .. }
```

Would help to clean-up the method, despite being pretty simple right now 🤔

Reply:
> However, I think it's also fine to clean it up and organize when we work specifically on empty states since we will also need to add additional cases and actions

Yeah, I'd say let's refactoring it later when we actually need it 👍